### ISSUE TITLE
Changing component props very fast makes it crash

### ISSUE THREAD

**Issue by louismahl** [diegomura/react-pdf#898]
**Describe the bug**
Changing component's state very fast makes it crash.

**To Reproduce**
I use redux with Firestore in the parent component. By the time it finishes loading all its data, it changes the props a few times (because it gets a first piece of data which allows it load another one right after): So the child component reloads very fast (maybe simultaneously?).

Here's the code i'm using for redux:
```
export default compose(
    connect(mapStateToProps),
    firestoreConnect((props) => {
        if (!props.auth.uid || !props.id) return [];

        return [{
            collection: 'data',
            doc: props.auth.uid,
            subcollections: [
                { ... }
            ],
            storeAs: '...',
        },{
            collection: 'data',
            doc: props.auth.uid,
            subcollections: [
                { ... },
                { ... }],
            storeAs: '...',
        },{
            collection: 'users',
            doc: props.auth.uid,
            storeAs: 'user',
        }]
    })
)(Details);
```

**Expected behavior**
Re-rendering each time the props change without crashing.

**Errors**
nbind.js:1272 Uncaught Error: Type mismatch
    at pushPointer (nbind.js:1272)

Uncaught TypeError: Cannot read property 'dereference' of undefined
    at __nbind_free_external (nbind.js:597)

The above error occurred in the <InternalBlobProvider> component:
    in InternalBlobProvider (created by PDFViewer)

Love this project btw!

**Desktop (please complete the following information):**
Chrome

**Comment by louismahl** [comment]
A quick fix is just to show the pdf viewer conditionally if all the data we're need loaded but this error could happen in other situations I believe

**Comment by mateo2181** [comment]
@louismahl did you solve this? I am getting the same error.

**Comment by louismahl** [comment]
> @louismahl did you solve this? I am getting the same error.

Hey ! You can quick fix this issue by doing so :

show pdf viewer conditionnally and use state data :

```
                                    {this.state.showPdf ?
                                        <PDFViewer style={{width: '100%', height: '700px'}}>
                                            <ReleveDeCharges yourData={this.state.yourData}/>
                                        </PDFViewer>
                                    : ""}
```

and then do this when your props data refreshes : 


```

componentDidUpdate(prevProps, prevState, snapshot) {
if (this.props !== prevProps) {
this.setState({
                        showPdf: false,
                    }, () => {
                        // Transfer props to state
                        this.setState({
                            yourData: this.props.yourData
                        }, () => {
                            this.setState({
                                showPdf: true,
                            })
                        })
                    })
    }
}
```

or use a button to show your pdf : 

`
<Button onClick={() => this.setState({showPdf: true})}>Show pdf</Button>
`

**Comment by mateo2181** [comment]
what version are you using? I need to use PDFDownloadLink component or pdf function to get blob info and download the file.

**Comment by louismahl** [comment]
You can do the same for PDFDownloadLink


                                    {this.state.showPdf ?
                                        <PDFDownloadLink document={<MyPDFComponent yourData={this.state.yourData}/>}
                                                         fileName={documentName}>
                                            {({ blob, url, loading, error }) => (loading ? 'Loading document...' : 'Download pdf')}
                                        </PDFDownloadLink>
                                    : ''}

**Comment by mateo2181** [comment]
thanks! are you using v. 1.6.0 or 2.0.0-beta.6?

**Comment by louismahl** [comment]
"@react-pdf/renderer": "^1.6.8"

**Comment by diegomura** [comment]
Should be fixed in 2.0
